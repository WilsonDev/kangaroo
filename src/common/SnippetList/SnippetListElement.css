.SnippetListElement {
  display: flex;
  justify-content: space-between;
  padding: 10px 15px 10px 15px;
}

.SnippetListElement-active {
  background-image: -webkit-gradient(linear, left top, right top, from(rgba(48, 64, 77, 0)), color-stop(40%, #30404d));
  background-image: linear-gradient(90deg, rgba(48, 64, 77, 0) 0, #30404d 40%);
}

.SnippetList--divider {
  height: 1px;
  margin-left: -50px;
  background-image: -webkit-gradient(linear,left top,right top,from(rgba(16,22,26,0)),color-stop(40%,rgba(16,22,26,.4)));
  background-image: linear-gradient(90deg,rgba(16,22,26,0) 0,rgba(16,22,26,.4) 40%);
}