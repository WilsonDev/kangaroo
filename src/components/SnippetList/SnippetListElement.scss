.SnippetListElement {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  padding: 10px 15px 10px 15px;
}

.SnippetListElement-active {
  background-image: var(--active-snippet);
}
