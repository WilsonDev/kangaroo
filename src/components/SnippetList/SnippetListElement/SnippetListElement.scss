@import "@blueprintjs/core/lib/scss/variables";

.SnippetListElement {
  height: var(--snippet-list-element-height);

  &.active {
    background-image: var(--active-snippet);

    .SnippetListElement--tag {
      box-shadow: 0 -0.5px 0px 1px var(--active-snippet-tag-border);
    }
  }
}

.SnippetListElement--content {
  position: relative;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  height: 100%;
  padding: 5px 15px 5px 15px;
  border-image: var(--divider) 1;
  border-bottom-style: solid;
  border-bottom-width: 1px;
}

.SnippetListElement--tag {
  position: absolute;
  width: 10px;
  height: 10px;
  right: 5px;
  border-radius: 50%;
  box-shadow: 0 -0.5px 0px 1px var(--background-color-secondary);
}

.SnippetListElement--source-icon {
  padding: 5px 10px;
}
