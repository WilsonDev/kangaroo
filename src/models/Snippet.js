export default class Snippet {
  constructor(args) {
    const { id, title, description, content } = args;

    this.id = id;
    this.title = title;
    this.description = description;
    this.content = content;
  }
}