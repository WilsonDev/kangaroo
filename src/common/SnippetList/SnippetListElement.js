import React from 'react';

import './SnippetListElement.css';

const SnippetListElement = ({element, current, handleChangeSnippet}) => {
  return (
    <div key={element.id}>
      <div className={`SnippetListElement ${current.id === element.id ? 'SnippetListElement-active' : null}`} onClick={() => handleChangeSnippet(element.id)}>
        {element.title}
      </div>
      <div className="SnippetList--divider" />
    </div>
  );
};

export default SnippetListElement;