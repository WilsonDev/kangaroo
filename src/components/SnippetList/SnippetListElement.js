import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { Button } from '@blueprintjs/core';

import Snippet, { sourceType } from '../../models/Snippet';

import './SnippetList.scss';
import './SnippetListElement.scss';

const SnippetListElement = ({
  element,
  currentlySelectedId,
  onChangeSnippet,
  onContextMenu
}) => {
  const handleClick = () => {
    if (currentlySelectedId !== element.id) {
      onChangeSnippet(element.id);
    }
  };

  const handleContextMenu = () => {
    handleClick();
    onContextMenu(element.id);
  };

  const listElementClass = classNames({
    'SnippetListElement': true,
    'SnippetListElement-active': currentlySelectedId === element.id,
    'bp3-text-muted': currentlySelectedId !== element.id
  });

  return (
    <div key={element.id}>
      <div
        className={listElementClass}
        onClick={handleClick}
        onContextMenu={handleContextMenu}
      >
        {element.title}
        {element.source === sourceType.GIST && (
          <Button icon="cloud" minimal={true} />
        )}
      </div>
      <div className="SnippetList--divider" />
    </div>
  );
};

SnippetListElement.propTypes = {
  element: PropTypes.instanceOf(Snippet),
  currentlySelectedId: PropTypes.number.isRequired,
  onChangeSnippet: PropTypes.func.isRequired,
  onContextMenu: PropTypes.func.isRequired
};

export default SnippetListElement;