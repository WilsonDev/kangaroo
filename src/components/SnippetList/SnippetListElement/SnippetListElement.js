import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { Icon } from '@blueprintjs/core';

import Snippet, { sourceType } from '../../../models/Snippet';

import './SnippetListElement.scss';

const SnippetListElement = ({
  element,
  currentlySelectedId,
  onChangeSnippet,
  onContextMenu
}) => {
  const handleClick = () => {
    if (currentlySelectedId !== element.id) {
      onChangeSnippet(element.id);
    }
  };

  const handleContextMenu = () => {
    handleClick();
    onContextMenu(element.id);
  };

  const listElementClass = classNames({
    'SnippetListElement': true,
    'active': currentlySelectedId === element.id,
    'bp3-text-muted': currentlySelectedId !== element.id
  });

  return (
    <div
      key={element.id}
      className={listElementClass}
      onClick={handleClick}
      onContextMenu={handleContextMenu}
    >
      <div className="SnippetListElement--content">
        {element.title}
        {element.source === sourceType.GIST && (
          <Icon className="SnippetListElement--source-icon" icon="cloud" />
        )}
      </div>
    </div>
  );
};

SnippetListElement.propTypes = {
  element: PropTypes.instanceOf(Snippet).isRequired,
  currentlySelectedId: PropTypes.number.isRequired,
  onChangeSnippet: PropTypes.func.isRequired,
  onContextMenu: PropTypes.func.isRequired
};

export default SnippetListElement;
