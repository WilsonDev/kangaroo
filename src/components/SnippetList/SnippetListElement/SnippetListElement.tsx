import React from 'react';
import classNames from 'classnames';
import { Icon } from '@blueprintjs/core';

import Snippet, { sourceType } from '../../../models/Snippet';

import './SnippetListElement.scss';

type SnippetListElementProps = {
  element: Snippet;
  currentlySelectedId: number | null;
  onChangeSnippet: (id: number) => void;
  onContextMenu: (id: number) => void;
}

const SnippetListElement = ({
  element,
  currentlySelectedId,
  onChangeSnippet,
  onContextMenu
}: SnippetListElementProps) => {
  const handleClick = () => {
    if (currentlySelectedId !== element.id) {
      onChangeSnippet(element.id);
    }
  };

  const handleContextMenu = () => {
    handleClick();
    onContextMenu(element.id);
  };

  const listElementClass = classNames({
    'SnippetListElement': true,
    'active': currentlySelectedId === element.id,
    'bp3-text-muted': currentlySelectedId !== element.id
  });

  return (
    <div
      key={element.id}
      className={listElementClass}
      onClick={handleClick}
      onContextMenu={handleContextMenu}
    >
      <div className="SnippetListElement--content">
        {element.title}
        {element.source === sourceType.GIST && (
          <Icon className="SnippetListElement--source-icon" icon="cloud" />
        )}
      </div>
    </div>
  );
};

export default SnippetListElement;
