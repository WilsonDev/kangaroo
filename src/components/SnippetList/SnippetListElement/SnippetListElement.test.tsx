import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { Button } from '@blueprintjs/core';
import SnippetListElement from './SnippetListElement';
import Snippet from 'models/Snippet';

configure({ adapter: new Adapter() });

describe('<SnippetListElement />', () => {
  const mockProps = {
    element: new Snippet({ id: 1, title: 'Test' }),
    currentlySelectedId: 1,
    onChangeSnippet: jest.fn(),
    onContextMenu: jest.fn(),
  };

  it('renders without crashing', () => {
    shallow(<SnippetListElement {...mockProps} />);
  });

  it('renders with <Button /> if sourceType is GIST', () => {
    const wrapper = shallow(<SnippetListElement {...mockProps} />);
    wrapper.contains(<Button />);
  });
});
