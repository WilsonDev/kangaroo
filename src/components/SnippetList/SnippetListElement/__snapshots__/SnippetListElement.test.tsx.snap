// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SnippetListElement /> matches snapshot 1`] = `
<div
  className="SnippetListElement active"
  onClick={[Function]}
  onContextMenu={[Function]}
>
  <div
    className="SnippetListElement--content"
  >
    <span
      className="bp3-text-overflow-ellipsis"
    >
      Test
    </span>
  </div>
</div>
`;
